src/options.c

```diff
@@ -497,35 +494,44 @@ void optionsParse(int argc, char *argv[])
             break;
         }
     }
-
-    /* Now the leftovers, which must be files */
-    while (optind < argc) {
-        /* If recursive is NOT set, but the only argument is a directory
-           name, we grab all the files in there, but not subdirs */
-        if (!opt.outputFile) {
-            optionsParseFileName(argv[optind++]);
-
-            const bool redirectChar = ( opt.outputFile[0] == '-'
-                                        && opt.outputFile[1] == '\0');
-            if (redirectChar) {
-                optionsParseFileName(getPathOfStdout());
-                opt.overwrite = 1;
-                opt.thumb = THUMB_DISABLED;
-            }
-        } else
-            warnx("unrecognised option %s", argv[optind++]);
+    argv += optind;
+
+    if (*argv) {
```

Comment:
I implemented with the semantics you suggest, but that has the weird semantics of an earlier argument (-F FILE) overriding the later non-option argument (FILE). Also, these weird semantics are more lines of code. 

I think it's better if I let the non-option argument override -F, is that fine?

Reply:
The convention for cli flags is that the later argument overrides earlier one. Which can be handy for wrapper scripts like this:

```sh
prog --flag=custom-default "$@"
```

But when both -F and non-option is specific (which in scrot's context have the same meaning) is weird to deal with, indeed. This gets even more weird when you realize that GNU getopt permutes the argument to bring flags before. E.g

```console
$ prog some-non-opt -F some-opt
```

On glibc systems, the `-F some-opt` will be parsed first where on musl it will stop option parsing after it finds the first non-opt.

I'll have to think about this a bit, more. I'm at the end of my awake hours too, so don't want to put much thought into it at this state :)